### PR TITLE
Fix performance of WinProcess.getEnvironmentVariables()

### LINKED ISSUE #68: obtaining the environmnet variables is inefficent
upstream of [JENKINS-67681](https://issues.jenkins.io/browse/JENKINS-67681)

The overhead in obtaining the process environemnt variables is too great resulting in a single no-op build on windows taking 700+ms just to enumerate all the processes looking for something to kill.

### PR DESCRIPTION
**Pull request by jtnord**
fixes #68 / [JENKINS-67681](https://issues.jenkins.io/browse/JENKINS-67681)

The performance of the code was no longer efficient in a Java5 world where `String.substring` involves memory copying to prevent holding onto larger strings preventing GC.

Benchmark showed the following:

```
# Run complete. Total time: 00:05:01

REMEMBER: The numbers below are just data. To gain reusable insights, you need to follow up on
why the numbers are the way they are. Use profilers (see -prof, -lprof), design factorial
experiments, perform baseline and negative tests that provide experimental control, make sure
the benchmarking environment is safe on JVM/OS/HW level, ask for reviews from the domain experts.
Do not assume the numbers tell you what you want them to tell.

Benchmark                                          Mode  Cnt  Score   Error  Units
ProcessEnvironmentVariableBenchmark.testNativeRaw  avgt    5  0.122 ± 0.024  ms/op
ProcessEnvironmentVariableBenchmark.testNew        avgt    5  0.135 ± 0.021  ms/op
ProcessEnvironmentVariableBenchmark.testOriginal   avgt    5  1.069 ± 0.112  ms/op
```

the java code used to take approximately .9ms per call and now takes approx 0.01ms

### CODE REVIEW

File: src/test/java/org/jvnet/winp/PlatformSpecificProcessTest.java [hunk 1]
```diff
@@ -106,7 +117,42 @@ public void getCommandLine_shouldNotFailIfTheProcessIsDead() throws Exception {
...
+    @Test
+    public void getEnvironmentVariables_shouldReturnCorrectValues() throws Exception {
+        Process p = spawnTestApp();
+        WinProcess wp = new WinProcess(p);
+        // spawned processes should inherit our environment variables
+        Map<String, String> inheritedEnv = System.getenv();
+        Map<String, String> processEnv = wp.getEnvironmentVariables();
+
+        // environment variable that start with = is just some funky stuff!
+        // and there are some that start with "=" that won't show up in set e.g. =ExitCode =CLINK.SCRIPTS
```
Comment (jtnord) [inline src/test/java/org/jvnet/winp/PlatformSpecificProcessTest.java:130]: I could not find any documentation on this (starting with `=`, but they exist and are funky!
so just excluded them from the test

e.g. 
```batch
set
echo %=ExitCode%
```

**Review by oleg-nenashev — APPROVED** [review]
LGTM, bumping to Java 1.8 is perfectly fine at this moment.
Note that I do not longer have release environment for Windows components and Maven Central. Please contact @batmat and @MarkEWaite to discuss who will take care of thios component, and I will help with getting permissions. Bonus points for a Jenkins devlist thread

File: src/main/java/org/jvnet/winp/WinProcess.java [hunk 2]
```diff
@@ -2,10 +2,7 @@
 
 import javax.annotation.CheckReturnValue;
 import java.lang.reflect.Field;
-import java.util.Comparator;
-import java.util.TreeMap;
-import java.util.Iterator;
-import java.util.logging.Level;
+import java.util.*;
```
[the remark this answers is not in the document]
Comment (jtnord) [reply src/main/java/org/jvnet/winp/WinProcess.java:5]: 👍  damn IDE defaults.  I switched to Intellij.

**Comment by jtnord** [comment]
> Note that I do not longer have release environment for Windows components and Maven Central. Please contact @batmat and @MarkEWaite to discuss who will take care of thios component, and I will help with getting permissions. Bonus points for a Jenkins devlist thread

https://groups.google.com/g/jenkinsci-dev/c/JmdP7Z0_ppc

Also cc @segrey

**Review by basil — APPROVED** [review]
Thanks for the PR!

**Comment by jtnord** [comment]
Thanks for merging Basil, I had forgotten about this one.

**Pull request merged**